### DISCUSSION TITLE
Presentation Slides

### DISCUSSION THREAD

**Discussion by pharringtonp19** [google-research/dex-lang#788] [Q&A] [answered]
I am really enjoying the language thus far -- I was wondering if it would be possible to convert the  "Live-updated notebook display" into presentation-like slides for teaching?

Thanks!

**Comment by dougalm** [comment]
We don't have a path for that, but it's a good idea! One option might be to add a format option to spit out the raw JSON data that's used to generate the HTML page, and then it would be easy enough to write something that uses pandoc to convert that to pdf.

Also, in case you didn't know, you can at least generate a self-contained static HTML page by running `dex script --outfmt html my_source_file.dx`.

**Reply by pharringtonp19** [reply]
@dougalm  I didn't know about `dex script --outfmt html my_source_file.dx`, so that's helpful.

I love how responsive katex is with `dex web`, so I was wondering whether it would be possible to keep the "live" aspect of the notebook, and somehow break up the notebook in a way so that only one section/header of the notebook appears at a time -- Thanks again for your thoughts

**Reply by dougalm** [reply]
I'm curious to know more about your use case. Do you want to see just one cell at a time because recompiling the other cells is slow? Or is it for presentation purposes? Or is it just to avoid getting distracted by the other cells? Would it be sufficient to have collapsible cells?

It almost sounds like you're using dex primarily as a latex authoring system. Not exactly its main intended purpose but we'll take it :) . I seriously dream that one day it'll be usable for developing program-specified diagrams, a bit like tikz but with a responsive interface and a more powerful language for abstracting patterns.

**Reply by pharringtonp19** [reply]
@dougalm Thanks for the response.

I teach an undergraduate Economics course. So when I present the material, the emphasis is mainly on the Economics and the coding is kept in the background. We use jax to compute gradients, and jaxopt to understand the sensitivity of our model predictions to our assumptions, but we leave it at that -- we certainly don't get into the finer details of jacobian vector products/ vector jacobian products, or things of that nature.

I am intrigued, though, by the functional programming aspect of dex. In  class, we try and think about most "things" in terms of functions (the optimal bundle as a function of one's income) and composition of functions (the utility of the optimal bundle) and so on. As I have learned more about Category Theory, Haskell, and Dex this semester, my sense is that a functional programming language better matches how we think about things in class, so I am considering transitioning to dex for this upcoming semester.

To get more to the point, I have no background in css/html/pandoc, or anything like that, so I feel a bit over my skis when sharing my thoughts. That said, if I had a magic wand, then in terms of using dex for a class lecture, it would be nice to be able to write down an optimization problem at the top of the slide (in katex), and call a function that solves the problem at the bottom of the slide. I love how clean, and responsive the current html output is, so if there's anyway to build that into a "deck of html slides", I would be all  for it. Thanks again for your time!

-- as to your question about cells (I assume that a cell is roughly equivalent to header/section?), I haven't figured out how to run one cell at a time, but I am sure that would make things nicer.

**Comment by apaszke** [comment, marked as answer]
This is a great idea! We should definitely add something like that!

**Reply by tscholak** [reply]
I'm currently working on a pandoc reader for dex for my blog that also evaluates the code... I can imagine that to be useful here.

**Reply by tscholak** [reply]
Still needs a bit of work, but the idea works: [Text.Pandoc.Readers.Dex](https://github.com/tscholak/website/blob/main/app/Text/Pandoc/Readers/Dex.hs). An example will be available shortly [here](https://tscholak.github.io/posts/hello-dex.html)
